transmogrifier/sources/transformer.py

```diff
@@ -118,6 +122,12 @@ def _etl_v2_next_iter_method(self) -> DatasetRecord:
             try:
                 transformed_record = self.transform(source_record)
                 timdex_record_id = transformed_record.timdex_record_id
+                transformed_record.timdex_provenance = timdex.TimdexProvenance(
+                    source=self.run_data["source"],
+                    run_date=self.run_data["run_date"],
+                    run_id=self.run_data["run_id"],
+                    run_record_offset=self.run_record_offset,
+                )
```

Comment:
I had considered locating this in the `Transformer.transform()` method, where the `TimdexRecord` instance is built, but opted not to for two reasons:

1. the `transform` method is still v1 and v2 compatible, whereas this would introduce feature flagging
2. the better of the two reasons, I think it's more associated with the orchestration of the run, the `__next__` iterator, etc., and makes sense in this part of the code vs the `transform` method which feels primarily focused on the metadata fields

Reply:
@ghukill To clarify, the provenance data will be added to TIMDEX records regardless of `ETL_VERSION`, right? It will be added to records generated by both v1 and v2?

Reply:
Not quite - the _model_ is updated to allow it ([see this related response](https://github.com/MITLibraries/transmogrifier/pull/233#discussion_r1929058408)), but only the v2 feature flagged method `_etl_v2_next_iter_method()` will actually add it to the record.